File: Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.php
<?php

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Integer
     */

    function max_height($root){
        $left = 0;
        $right = 0;

        if($root->left != null){
            $left = 1+max_height($root->left);
        }
        if($root->right != null){
            $right = 1+max_height($root->right);
        }
        if($root->left == null && $root->right == null){
            return 0;
        }

        return max($left, $right);
    }

    function maxDepth($root) {

        if($root==null){
            return 0;
        }
        return $this->max_height($root) + 1;
    }
}

?>

File: Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
        
        def max_height(root):
            left = 0
            right = 0

            if root.left:
                left = 1+max_height(root.left)
            if root.right:
                right = 1+max_height(root.right)
            if root.left is None and root.right is None:
                return 0
        
            return max(left, right)
        
        return max_height(root)+1